File: Gemini_Flash/Java/Facade/round2/test/test_car2.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class CarTest {

    @Test
    void testStartEngine() {
        Car car = new Car();
        car.startEngine();
        // ตรวจสอบข้อความที่ถูกพิมพ์ออกไป
        String expectedOutput = "เครื่องยนต์สตาร์ท\n";
        assertEquals(expectedOutput, getOutputFromSystemOut());
    }

    @Test
    void testStopEngine() {
        Car car = new Car();
        car.stopEngine();
        String expectedOutput = "เครื่องยนต์ดับ\n";
        assertEquals(expectedOutput, getOutputFromSystemOut());
    }

    @Test
    void testTurnOnLights() {
        Car car = new Car();
        car.turnOnLights();
        String expectedOutput = "ไฟหน้าเปิด\n";
        assertEquals(expectedOutput, getOutputFromSystemOut());
    }

    @Test
    void testTurnOffLights() {
        Car car = new Car();
        car.turnOffLights();
        String expectedOutput = "ไฟหน้าปิด\n";
        assertEquals(expectedOutput, getOutputFromSystemOut());
    }

    @Test
    void testTurnLeft() {
        Car car = new Car();
        car.turnLeft();
        String expectedOutput = "เลี้ยวซ้าย\n";
        assertEquals(expectedOutput, getOutputFromSystemOut());
    }

    @Test
    void testTurnRight() {
        Car car = new Car();
        car.turnRight();
        String expectedOutput = "เลี้ยวขวา\n";
        assertEquals(expectedOutput, getOutputFromSystemOut());
    }

    @Test
    void testApplyBrakes() {
        Car car = new Car();
        car.applyBrakes();
        String expectedOutput = "เบรก\n";
        assertEquals(expectedOutput, getOutputFromSystemOut());
    }

    @Test
    void testReleaseBrakes() {
        Car car = new Car();
        car.releaseBrakes();
        String expectedOutput = "ปลดเบรก\n";
        assertEquals(expectedOutput, getOutputFromSystemOut());
    }

    // วิธีการช่วยในการรับข้อความจาก System.out
    private String getOutputFromSystemOut() {
        // นี่คือตัวอย่างการรับข้อความจาก System.out 
        // คุณอาจต้องปรับเปลี่ยนวิธีการนี้ตามความเหมาะสม
        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
        System.setOut(new PrintStream(outputStream));
        return outputStream.toString();
    }
}

File: Copilot/Python/Facade/round2/test/test_car2.py
# test_car_facade.py
import pytest
from unittest.mock import MagicMock
from car_facade import CarFacade

@pytest.fixture
def car_facade():
    car_facade = CarFacade()
    car_facade.engine = MagicMock()
    car_facade.lights = MagicMock()
    car_facade.steering_wheel = MagicMock()
    car_facade.brakes = MagicMock()
    return car_facade

def test_start_engine(car_facade):
    car_facade.start_engine()
    car_facade.engine.start.assert_called_once()

def test_stop_engine(car_facade):
    car_facade.stop_engine()
    car_facade.engine.stop.assert_called_once()

def test_turn_on_lights(car_facade):
    car_facade.turn_on_lights()
    car_facade.lights.turn_on.assert_called_once()

def test_turn_off_lights(car_facade):
    car_facade.turn_off_lights()
    car_facade.lights.turn_off.assert_called_once()

def test_turn_left(car_facade):
    car_facade.turn_left()
    car_facade.steering_wheel.turn_left.assert_called_once()

def test_turn_right(car_facade):
    car_facade.turn_right()
    car_facade.steering_wheel.turn_right.assert_called_once()

def test_apply_brakes(car_facade):
    car_facade.apply_brakes()
    car_facade.brakes.apply.assert_called_once()

def test_release_brakes(car_facade):
    car_facade.release_brakes()
    car_facade.brakes.release.assert_called_once()